### PR TITLE
Nutritionfacts i18n

### PR DESCRIPTION
**Pull request by indirectlylit**
### Summary

* add a script to extract internationalized update notification JSON blobs
* add basic documentation on how to verify and send messages

### Reviewer guidance

check it out

### References

ref: https://www.notion.so/learningequality/Kolibri-upgrade-notification-process-design-6e8ca1ebe09b4c2b9b3d9b85c3308155

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: docs/release_process.rst [hunk 1]
```diff
@@ -248,3 +248,65 @@ Wrap-up
...
+
+
+Send upgrade notifications
+--------------------------
+
+Wait about 3 business days after communications are published to see if any issues are reported. Afterwards, we can send upgrade notifications through the Nutrition Facts telemetry server.
+
+* `Log in to the telemetry server <http://telemetry.learningequality.org/account/login/google-oauth2/?next=/admin/>`__ using your Learning Equality Google Apps account
+* Create a new Message object in the server and set it to 'Staged'
+* Set the link URL to ``https://learningequality.org/r/upgrade_kolibri``
```
Comment (jonboiser) [inline docs/release_process.rst:261]: Can add more details here like

1. Click the "+ Add" link next to "Messages"
1. In the "Version range" field, enter a valid semver range (e.g. >=0.12.0)
1. In the "Status" dropdown, select the "Staged" option

File: build_tools/i18n/crowdin.py [hunk 2]
```diff
@@ -541,4 +568,6 @@ def main():
 
 
 if __name__ == "__main__":
+
+    print("JSEHFKSJDFJKSDFDF???????/")
```
Comment (jonboiser) [inline build_tools/i18n/crowdin.py]: This looks like it's left over.

File: build_tools/i18n/nutritionfacts_i18n.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+"""
+For usage instructions, see:
+    https://kolibri-dev.readthedocs.io/en/develop/references/i18n.html
```
Comment (jonboiser) [inline build_tools/i18n/nutritionfacts_i18n.py]: This should just be https://kolibri-dev.readthedocs.io/en/develop/i18n.html ( no `/references`)

Comment (jonboiser) [reply build_tools/i18n/nutritionfacts_i18n.py]: And I don't think there's docs for this function yet.

File: docs/release_process.rst [hunk 4]
```diff
@@ -248,3 +248,65 @@ Wrap-up
...
+
+  python build_tools/i18n/nutritionfacts_i18n.py
+
+You can also specify specific string IDs for the ``title``, ``msg``, and ``link_text``, e.g.:
+
+.. code-block:: bash
+
+  python build_tools/i18n/nutritionfacts_i18n.py --message UpdateNotification.upgradeMessage0124
+
+This will output a JSON blob like:
```
Comment (jonboiser) [inline docs/release_process.rst:275]: Maybe add a suggestion to redirect the out put to a text file or the clipboard

```
 python build_tools/i18n/nutritionfacts_i18n.py --message UpdateNotification.upgradeMessage0124 >> ~/Documents/messages.txt
```

File: build_tools/i18n/nutritionfacts_i18n.py (hunk 3 above)
Comment (indirectlylit) [reply build_tools/i18n/nutritionfacts_i18n.py]: the docs are in this PR! :)

File: build_tools/i18n/crowdin.py (hunk 2 above)
Comment (indirectlylit) [reply build_tools/i18n/crowdin.py]: lol

File: build_tools/i18n/nutritionfacts_i18n.py (hunk 3 above)
Comment (indirectlylit) [reply build_tools/i18n/nutritionfacts_i18n.py]: ah, I meant to reference https://kolibri-dev.readthedocs.io/en/develop/release_process.html

**Comment by indirectlylit** [comment]
updated with suggestions

File: docs/release_process.rst [hunk 5]
```diff
@@ -248,3 +248,68 @@ Wrap-up
...
+      "msg": "Una nueva versión de Kolibri está disponible.",
+      "title": "Actualización disponible"
+    },
+    //...
+  }
+
+
+You can `redirect this output to a file <https://askubuntu.com/questions/420981/how-do-i-save-terminal-output-to-a-file>`_ (Bash) or `pipe it to the clipboard <https://stackoverflow.com/questions/1753110/>`_ (Mac)
+
+Set Kolibri's ``RUN_MODE`` to ``staged-msgs`` to receive staged messages. Test that all languages are displayed correctly.
```
Comment (jonboiser) [inline docs/release_process.rst:311]: How do you do this again? In my zsh history, I have `launchctl setenv KOLIBRI_RUN_MODE "jb"`. Does that work in Linux too?

Comment (jonboiser) [reply docs/release_process.rst:311]: Or just `export KOLIBRI_RUN_MODE="staged-msgs"` in the session?

File: docs/release_process.rst [hunk 6]
```diff
@@ -248,3 +248,68 @@ Wrap-up
...
+
+Next, emulate different versions and ensure that the semver conditional logic is being processed correctly. Set ``RUN_MODE`` to something like ``staged-msgs-ver-0.12.3`` to emulate version 0.12.3, for example.
```
Comment (jonboiser) [inline docs/release_process.rst:313]: How does this emulate a different version? Does it parse the `RUN_MODE` to infer the version?

Comment (jonboiser) [reply docs/release_process.rst:313]: That's how nutrition facts seems to work. Maybe could clarify that by saying that in general set KOLIBRI_RUN_MODE to `staged-msgs-ver-<semver version number>`

**Review by jonboiser — APPROVED** [review]

**Comment by indirectlylit** [comment]
new PR with additional documentation updates based on feedback: https://github.com/learningequality/kolibri/pull/5707

**Pull request merged**